[Vishnu Sundaresan] iP

juliussneezer04 opened the pull request:
# Duke
## by Juliussneezer04
>“By Failing to prepare, you are preparing to fail.” 
>– Benjamin Franklin

Duke is the ultimate command-line chat bot for task management and reminders 🗓️ 
Duke can help you track 📌 
* your weekly tasks 
* your deadlines
* your events
in one simple ~~graphical-user interface~~ **command-line interface** 💻 

Duke currently supports
1. Task Creation ➕ 
2. Task Deletion ➖ 
3. Task Searching 🔍 
4. Task Access 📔 

Duke uses simple code like as shown here `this.dukeUi.welcomeUser();`

and also complex code like here
```java
FileWriter fw = new FileWriter(dukeData);
BufferedWriter dukeWriter = new BufferedWriter(fw);
for (Entry entry : entries) {
  String nextEntry = entry.saveString();
  dukeWriter.write(nextEntry);
  dukeWriter.write("\n");
}
```

*Updated Features list*
- [x] Task Creation & Deletion
- [x] Task Searching 
- [ ] Graphical User Interface

Commit:
Add Gradle support

juliussneezer04 committed:
Level-1 Completed.

juliussneezer04 committed:
Level-2 Completed.

juliussneezer04 committed:
Level-3 Completed.

juliussneezer04 committed:
Level-4 Completed.

juliussneezer04 committed:
Level-4 Updated.

juliussneezer04 committed:
A-TextUiTesting Completed.

juliussneezer04 committed:
Level-5 Completed.

juliussneezer04 committed:
Level-6 Completed.

juliussneezer04 committed:
Saving implemented.

juliussneezer04 committed:
Implemented Date

juliussneezer04 committed:
Merge branch 'branch-Level-7'

juliussneezer04 committed:
Merge branch 'branch-Level-8'

juliussneezer04 committed:
Merge branch-level-8 conflicts

juliussneezer04 committed:
A-MoreOOP

juliussneezer04 committed:
A-Packages

juliussneezer04 committed:
A-JUnit

juliussneezer04 committed:
A-Jar

juliussneezer04 committed:
A-JavaDoc

juliussneezer04 committed:
A-CodingStandard

juliussneezer04 committed:
branch-Level-9

juliussneezer04 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java

juliussneezer04 committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Entry.java
#	src/main/java/duke/EntryList.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java

kevin9foong commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
```
Hi Vishnu, a small nit but you might want to consider adding some header Javadocs for this class

kevin9foong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,90 @@
+    private void parseEntry(ArrayList<String> terms) {
+        if (!terms.isEmpty()) {
+            StringBuilder entry = new StringBuilder();
+            //Combine All Strings Until End of List or '/' character is found
```
There should be a spacing between the comment and the `//` based on the examples given in the coding guide.

juliussneezer04 replied:
Resolved, thanks!

kevin9foong commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    /**
+     * Constructor for Todo.
```
Seems like the javadocs description for the parameter is missing for this JavaDocs header comment

juliussneezer04 replied:
resolved, thanks!

kevin9foong commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * Prints out farewell message.
+     */
+    public void goodByeUser() {
```
This method's name could potentially be improved as method names should be verbs. Perhaps something like 'printGoodByeMessage'.

juliussneezer04 replied:
resolved, thanks!

kevin9foong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,127 @@
+    private final Storage dukeStorage;
+    private EntryList entries;
+
+    private final String TERMINATION_COMMAND = "bye";
```
I like how the constants are named using uppercase and underscores, makes the styling very organized.

kevin9foong commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,65 @@
+     * @return Boolean corresponding to Event's length.
+     */
+    @Override
+    public boolean isEmpty() { return super.isEmpty() || !event.isAfter(LocalDate.now()); }
```
You might want to consider using egyptian braces rather than a single line method body as mentioned in the coding standard.

juliussneezer04 replied:
Ah thanks! Intellij collapsed my braces automatically unfortunately

kevin9foong reviewed (commented):
Hi Vishnu, I've been assigned to leave a pull request review on your code for coding style. I've made a few minor comments some things I've found. Hopefully it helps you out! 👍

juliussneezer04 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Todo.java

juliussneezer04 committed:
Un-doing merge conflicts from Branch-level-8

juliussneezer04 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Nature711 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+    /**
+     * Constructor for Deadline.
+     */
+    Deadline() {
```
Is it better to add a "public" access modifier to the constructor?

Nature711 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,66 @@
+     * @return Boolean corresponding to Deadline's length.
+     */
+    @Override
+    public boolean isEmpty() { return super.isEmpty() && !deadline.isAfter(LocalDate.now()); }
```
Is it better to put the method body into a separate line?

juliussneezer04 replied:
sigh, the worries of intellij automatic code styling...

Nature711 commented on src/main/java/duke/EntryList.java:
```diff
@@ -0,0 +1,112 @@
+        for (int i = 0; i < len; i++) {
+            Entry currentEntry = super.get(i);
+            if (currentEntry.contains(keyword)) {
+                if (count == 1) {
```
Maybe try to combine the condition check "count == 1" with the line above?

juliussneezer04 replied:
Over here I am specifically calling foundMatches , when count is 1, to display a heading line for the first found match. Hence the nested if condition.

Nature711 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,102 @@
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.BufferedWriter;
```
Seems like the lexicographical ordering for import statement is a bit wrong.

Nature711 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,102 @@
+     * @param entry THe newly-added entry.
+     * @param id The #id of the newly-added entry.
+     */
+    public void addEntry (Entry entry, int id) {
```
Seems like you don't need a white space between the method name and the parentheses.

Nature711 reviewed (commented):
LGTM except for some minor coding standard issues :D

juliussneezer04 committed:
PR Review Resolutions

juliussneezer04 committed:
A-Gradle

juliussneezer04 committed:
A-Gradle runs unit tests.

juliussneezer04 committed:
Java-FX dependencies

juliussneezer04 committed:
Basic GUI for Duke.

juliussneezer04 committed:
Assertion statements are not utilized.

Assertion statements are efficient mechanisms in catching bugs and
exceptions early.

We will include assertion statement, which will be useful to
Duke for the aforementioned reason.

The assertions have been placed in the Parser class to facilitate
exception handling.

juliussneezer04 committed:
Merge pull request #1 from juliussneezer04/branch-A-Assertions

Assertion statements are not utilized.

juliussneezer04 committed:
Merge branch 'master' of https://github.com/juliussneezer04/ip

juliussneezer04 committed:
Adding a help page.

There is no help page now.

The help page will help users:
* See available commands
* Learn Duke syntax

Let's add the help command so users can easily all commands at once.

juliussneezer04 committed:
Code Quality is poor.

We can improve code quality to make collaborations smooth and the code
look more understandable.

Let's modify the code according to the Java Coding standards.

juliussneezer04 committed:
Merge pull request #2 from juliussneezer04/branch-C-Help

Adding a help page.

juliussneezer04 committed:
Merge branch 'master' into branch-A-CodeQuality

juliussneezer04 committed:
Merge pull request #3 from juliussneezer04/branch-A-CodeQuality

A-CodeQuality

juliussneezer04 committed:
FXML is not incorporated.

FXML helps the separation of Ui from Logic code.

Let's add FXML.

juliussneezer04 committed:
FXML is fully incorporated.

FXML simplifies UI design.
Adding FXML helps us abstract design from logic.

juliussneezer04 committed:
User Guide Updated.

User Guide was lacking.
It is updated to include project details.

juliussneezer04 committed:
Set theme jekyll-theme-minimal

juliussneezer04 committed:
Set theme jekyll-theme-tactile

juliussneezer04 committed:
Releasing latest jar.

v0.3 jar file is to be released.
